File: algorithms/1572/main.scala
object Solution {
    def diagonalSum(mat: Array[Array[Int]]): Int = {
        val sum = (for (i <- 0 to mat.length/2-1) yield {
            val j = mat.length-1
            mat(i)(i) + mat(i)(j-i) + mat(j-i)(i) + mat(j-i)(j-i)
        }).sum
        if (mat.length % 2 == 1) sum + mat(mat.length/2)(mat.length/2) else sum
    }
}

File: algorithms/1572/main.go
func diagonalSum(mat [][]int) int {
    matLength := len(mat)
    l := matLength-1
    index := matLength/2
    sum := 0
    for i := 0 ; i < index ; i++ {
        j := l-i
        sum += mat[i][i] + mat[i][j] + mat[j][i] + mat[j][j]
    }
    if matLength % 2 == 1 {
        sum += mat[index][index]
    }
    return sum
}
